src/main/java/org/sputnikdev/bluetooth/gattparser/spec/BluetoothGattSpecificationReader.java

```diff
@@ -221,6 +226,54 @@ public void loadExtensionsFromFolder(String path) {
         readCharacteristics(getFilesFromFolder(characteristicsFolderName));
     }
 
+    private static URL specResourceURL(URL original, String characteristicType) throws MalformedURLException {
+        String oldfile = original.getFile();
+        int lastslash = oldfile.lastIndexOf('/');
+        String newfile = oldfile;
+        if (lastslash >= 0) {
+            newfile = oldfile.substring(0, lastslash);
+        }
+
+        newfile = newfile + "/" + characteristicType + ".xml";
+        return new URL(
+            original.getProtocol(),
+            original.getHost(), 
+            original.getPort(),
+            newfile
+        );
+    }
+
+    private Map<String, URL> catalogToURLs(URL serviceRegistry, Map<String, String> xmlEntry) {
+        Map<String, URL> processed = new HashMap();
```

Comment:
Looks like you are using Eclipse, it does not complain about type casting in that case, however IDEA does :)

effectively Map<String, URL> is not compatible with HashMap, but the following is:
```
Map<String, URL> processed = new HashMap<String, URL>();
```
or short:

```
Map<String, URL> processed = new HashMap<>();
```

Reply:
Just plain vim. Rewrote & will try to remember.